### PR TITLE
Selenium test for Data Use Aware Search [GAWB-2928]

### PR DESCRIPTION
**Pull request by aednichols**
~~Exists only to trigger tests for now.~~

Addresses:
- RP modal opens/closes properly
- RP modal shows appropriate breadcrumbs
---
- [ ] **Submitter**: Include the JIRA issue number in the PR description
- [ ] **Submitter**: Check documentation and code comments. Add explanatory PR comments if helpful.
- [ ] **Submitter**: If you changed a URL that is used elsewhere (e.g. in an email), comment about where it is used and ensure the dependent code is updated.
- [ ] **Submitter**: JIRA ticket checks:
  * Acceptance criteria exists and is met
  * Note any changes to implementation from the description
  * To Demo flag is set
  * Release Summary is filled out, if applicable
  * Add notes on how to QA
- [ ] **Submitter**: Update RC_XXX release ticket with any config or environment changes necessary
- [ ] **Submitter**: Update FISMA documentation if changes to:
  * Authentication
  * Authorization
  * Encryption
  * Audit trails
- [ ] **Submitter**: If you're adding new libraries, sign us up to security updates for them
- [ ] **Submitter**: If you're adding new automated UI tests, review the test plan with QA
* Review cycle:
  * LR reviews
  * Rest of team may comment on PR at will
  * **LR assigns to submitter** for feedback fixes
  * Submitter rebases to develop again if necessary
  * Submitter makes further commits. DO NOT SQUASH
  * Submitter updates documentation as needed
  * Submitter **reassigns to LR** for further feedback
- [ ] **TL** sign off
- [ ] **LR** sign off
- [ ] **Product Owner** sign off
- [ ] **Submitter**: Verify all tests go green, including CI tests and automated UI tests.
- [ ] **Submitter**: Squash commits and merge to develop. If adding test code, merge application code and test code at the same time.
- [ ] **Submitter**: Delete branch after merge
- [ ] **Submitter**: **Test this change works on dev environment after deployment**. YOU own getting it fixed if dev isn't working for ANY reason!
- [ ] **Submitter**: Mark JIRA issue as resolved once this checklist is completed

### CODE REVIEW

File: automation/src/test/scala/org/broadinstitute/dsde/firecloud/page/library/DataLibraryPage.scala [hunk 1]
```diff
@@ -16,6 +16,10 @@ class DataLibraryPage(implicit webDriver: WebDriver) extends BaseFireCloudPage
 
   private val LibraryTable = Table("library-table")
   private val searchField = SearchField("library-search-input")
+  private val rpModalLink = Link("show-research-purpose-modal")
```
Comment (aednichols) [inline automation/src/test/scala/org/broadinstitute/dsde/firecloud/page/library/DataLibraryPage.scala:19]: I was thinking about making a new class to represent the modal separately, but it's so simple I decided not to for now

Comment (MattPutnam) [reply automation/src/test/scala/org/broadinstitute/dsde/firecloud/page/library/DataLibraryPage.scala:19]: Sometime soon, I'm going to make Modal a Component a la Collapse. For now this is fine.

**Comment by aednichols** [comment]
Failing tests are in `AuthDomainSpec` and `FreeTrialSpec`. I think at this point `jenkins retest` is kind of an antipattern and just burns limited execution slots, so I'll hold off until we are more stable

**Comment by aednichols** [comment]
Wow, it parsed it right out of that comment. TIL

**Comment by aednichols** [comment]
https://github.com/broadinstitute/firecloud-ui/pull/1140 may help the particular way that `FreeTrialSpec` blew up in the above run

File: automation/src/test/scala/org/broadinstitute/dsde/firecloud/page/library/DataLibraryPage.scala (hunk 1 above)
Comment (ahaessly) [reply automation/src/test/scala/org/broadinstitute/dsde/firecloud/page/library/DataLibraryPage.scala:19]: we already have an OKCancelModal abstract class. org.broadinstitute.dsde.firecloud.page.Modals.scala

File: automation/src/test/scala/org/broadinstitute/dsde/firecloud/test/library/DataUseAwareSearchSpec.scala [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+      withSignIn(user) { _ =>
+        val page = new DataLibraryPage().open
+
+        page.openResearchPurposeModal()
+
+        page.isShowingResearchPurposeModal shouldBe true
+
+        page.dismissResearchPurposeModal()
+
+        page.isShowingResearchPurposeModal shouldBe false
```
Comment (davidangb) [inline automation/src/test/scala/org/broadinstitute/dsde/firecloud/test/library/DataUseAwareSearchSpec.scala:30]: another test case: for good measure, open and close it twice, to ensure that after closing it is re-openable. Think that's overkill?

**Review by davidangb — APPROVED** [review]
one optional comment inline; if that's all you change I don't need to re-review. Thanks! 👍

**Review by ahaessly — APPROVED** [review]
👍

File: automation/src/test/scala/org/broadinstitute/dsde/firecloud/test/library/DataUseAwareSearchSpec.scala (hunk 2 above)
Comment (aednichols) [reply automation/src/test/scala/org/broadinstitute/dsde/firecloud/test/library/DataUseAwareSearchSpec.scala:30]: There's at least one more PR coming from this ticket, so I'll probably roll this into that one.

**Pull request merged**